pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TimeSeriesIntegrationTest.java

```diff
@@ -0,0 +1,306 @@
+        validator.validate(ts, val, row);
+      }
+    }
+  }
+
+  @FunctionalInterface
+  interface TimeSeriesValidator {
+    void validate(long timestamp, long value, JsonNode row);
+  }
+
+  @Override
+  protected void overrideBrokerConf(PinotConfiguration brokerConf) {
+    addTimeSeriesConfigurations(brokerConf);
+  }
+
+  @Override
+  protected void overrideServerConf(PinotConfiguration serverConf) {
+    addTimeSeriesConfigurations(serverConf);
+  }
+
+  @Override
+  public String getTableName() {
+    return DEFAULT_TABLE_NAME;
+  }
+
+  @Override
+  public long getCountStarResult() {
+    return 1000L;
+  }
+
+  @Override
+  public Schema createSchema() {
+    return new Schema.SchemaBuilder().setSchemaName(getTableName())
+      .addSingleValueDimension(TS_COLUMN, FieldSpec.DataType.LONG)
+      .addSingleValueDimension(CITY_COLUMN, FieldSpec.DataType.LONG)
+      .addSingleValueDimension(PLATFORM_COLUMN, FieldSpec.DataType.STRING)
+      .addSingleValueDimension(REFERRAL_COLUMN, FieldSpec.DataType.BOOLEAN)
+      .addSingleValueDimension(VIEWS_COLUMN, FieldSpec.DataType.LONG)
+      .build();
+  }
+
+  private org.apache.avro.Schema.Field createAvroField(String name, org.apache.avro.Schema.Type type) {
+    return new org.apache.avro.Schema.Field(name, org.apache.avro.Schema.create(type), null, null);
+  }
+
+  public File createAvroFile()
+    throws Exception {
+    org.apache.avro.Schema avroSchema = org.apache.avro.Schema.createRecord("myRecord", null, null, false);
+    avroSchema.setFields(ImmutableList.of(
+      createAvroField(TS_COLUMN, org.apache.avro.Schema.Type.LONG),
+      createAvroField(CITY_COLUMN, org.apache.avro.Schema.Type.LONG),
+      createAvroField(PLATFORM_COLUMN, org.apache.avro.Schema.Type.STRING),
+      createAvroField(REFERRAL_COLUMN, org.apache.avro.Schema.Type.BOOLEAN),
+      createAvroField(VIEWS_COLUMN, org.apache.avro.Schema.Type.LONG)
+    ));
+
+    File avroFile = new File(_tempDir, "data.avro");
+    try (DataFileWriter<GenericData.Record> writer = new DataFileWriter<>(new GenericDatumWriter<>(avroSchema))) {
+      writer.create(avroSchema, avroFile);
+      for (int i = 0; i < getCountStarResult(); i++) {
+        writer.append(getRecord(avroSchema, i));
+      }
+    }
+    return avroFile;
+  }
+
+  private static GenericData.@NotNull Record getRecord(org.apache.avro.Schema avroSchema, int i) {
+    GenericData.Record record = new GenericData.Record(avroSchema);
+    // Do not set DATA_START_TIME_SEC for easier assertion of values.
+    record.put(TS_COLUMN, (DATA_START_TIME_SEC + 1 + i) * 1000L);
+    record.put(CITY_COLUMN, i % 5);
+    record.put(PLATFORM_COLUMN, PLATFORMS[i % PLATFORMS.length]);
+    record.put(REFERRAL_COLUMN, (i % 2) == 0);
+    // Alternate between VIEWS_MIN_VALUE and VIEWS_MAX_VALUE.
+    record.put(VIEWS_COLUMN, VIEWS_MIN_VALUE + (VIEWS_MAX_VALUE - VIEWS_MIN_VALUE) * (i % 2));
+    return record;
+  }
+
+  @BeforeClass
+  public void setUp()
```

Comment:
nit: suggest keeping public methods at the top for readability

Reply:
(this was marked as resolved looks like)

recommend keeping setUp at the top as is standard pattern